### DISCUSSION TITLE
Highlighting match is often misaligned

### DISCUSSION THREAD

**Discussion by ijsnow** [community/community#44809] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

Hello! Love the new code search.

I've just noticed more than once that the match highlighting is often not exactly highlighting what was searched in the various code views. See this photo where one is off by one and the other is off by a few.

<img width="1119" alt="Screen Shot 2023-01-19 at 10 21 26 AM" src="https://user-images.githubusercontent.com/11557758/213528219-ed6b3d58-83b3-49b8-8423-0aaad408ac5c.png">

**Comment by jsoref** [comment]
Do you have an example url where this still happens? (Or the commit tree for which the above was a problem? -- I can't get what I think is your query to turn up anything and don't want to bisect your repo looking for it... https://github.com/search?q=repo%3Aalacritty%2Falacritty%20vte%3A%3APerform&type=code)

Here's the closest I can find atm:
https://github.com/search?q=repo%3Aalacritty%2Falacritty+serde%3A%3Ade%3A%3AVisitor&type=code
<img width="1377" alt="image" src="https://github.com/community/community/assets/2119212/03581f29-a01f-4f46-aa51-3798f4e22a3b">
